### PR TITLE
Review

### PR DESCRIPTION
**Pull request by liadomide**
Merge latest changes for having a base-line in master.
PSE included in this commit.
Drastic review will follow soon.

### CODE REVIEW

File: tvb_epilepsy/tvb_api/simulator_tvb.py [hunk 1]
```diff
@@ -142,38 +162,54 @@ def prepare_for_h5(self, settings):
...
+                                                                             self.simTVB.good_history_shape[0])
 
 ###
 # A helper function to make good choices for simulation settings, noise and monitors
 ###
 
-def setup_simulation(model_name, hypothesis, dt, sim_length, monitor_period, zmode=numpy.array("lin"), scale_time=1,
-                     noise_instance=None, noise_intensity=None,
+def setup_simulation(hypothesis, head_connectivity, dt, sim_length, monitor_period, model_name="EpileptorDP",
+                     zmode=numpy.array("lin"), scale_time=1, noise_instance=None, noise_intensity=None,
```
Comment (maedoc) [inline tvb_epilepsy/tvb_api/simulator_tvb.py]: It's a bug-waiting-to-happen to use a mutable data structure as a keyword default; better to use `None` and, below, `self.zmode = zmode or numpy.array("lin")`.

File: tvb_epilepsy/base/equations.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import numpy
-from numpy import array, empty, empty_like, ones, eye, zeros, multiply, dot, power, divide, sum, exp, reshape, \
+from numpy import array, empty, empty_like, ones, eye, zeros, multiply, dot, power, divide, sum, exp, reshape, tile, \
```
Comment (maedoc) [inline tvb_epilepsy/base/equations.py]: Is there a reason here not to use the conventional `import numpy as np`?

File: tvb_epilepsy/base/equations.py [hunk 3]
```diff
@@ -258,8 +258,8 @@ def eqtn_jac_fz_2d(x1, z, tau1, tau0, zmode=array("lin"), z_pos=True, K=None, w=
 
     # Assuming that wii = 0
     jac_x1 += multiply(K, sum(w, 1))
-    jac_x1 = diag(jac_x1.flatten()) - multiply(repeat(reshape(K, (x1.size, 1)), 3, axis=1), w)
-    jac_x1 *= repeat(reshape(tau, (x1.size, 1)), 3, axis=1)
+    jac_x1 = diag(jac_x1.flatten()) - multiply(repeat(reshape(K, (x1.size, 1)), x1.size, axis=1), w)
```
Comment (maedoc) [inline tvb_epilepsy/base/equations.py]: These `repeat`s and `reshape`s would be far cleaner if they were written as regular arithmetic and broadcasting operations.

File: tvb_epilepsy/tvb_api/simulator_tvb.py (hunk 1 above)
Comment (liadomide) [inline tvb_epilepsy/tvb_api/simulator_tvb.py]: Fully agree. We will be refactoring these in the next steps

**Pull request merged**